ArgusProbe_Beamlogic.py

```diff
@@ -227,6 +228,284 @@ def _get_ntp_timestamp():
+    0xd94c, 0xc96d, 0xf90e, 0xe92f, 0x99c8, 0x89e9, 0xb98a, 0xa9ab,
+    0x5844, 0x4865, 0x7806, 0x6827, 0x18c0, 0x08e1, 0x3882, 0x28a3,
+    0xcb7d, 0xdb5c, 0xeb3f, 0xfb1e, 0x8bf9, 0x9bd8, 0xabbb, 0xbb9a,
+    0x4a75, 0x5a54, 0x6a37, 0x7a16, 0x0af1, 0x1ad0, 0x2ab3, 0x3a92,
+    0xfd2e, 0xed0f, 0xdd6c, 0xcd4d, 0xbdaa, 0xad8b, 0x9de8, 0x8dc9,
+    0x7c26, 0x6c07, 0x5c64, 0x4c45, 0x3ca2, 0x2c83, 0x1ce0, 0x0cc1,
+    0xef1f, 0xff3e, 0xcf5d, 0xdf7c, 0xaf9b, 0xbfba, 0x8fd9, 0x9ff8,
+    0x6e17, 0x7e36, 0x4e55, 0x5e74, 0x2e93, 0x3eb2, 0x0ed1, 0x1ef0,
+    )
+
+    def __init__(self, txMqttThread, serialport,baudrate):
+
+        # store params
+        self.txMqttThread            = txMqttThread
+        self.serialport              = serialport
+        self.baudrate                = baudrate
+
+        # local variables
+        self.serialHandler           = None
+        self.goOn                    = True
+        self.pleaseConnect           = True
+        self.dataLock                = threading.RLock()
+
+        # hdlc frame parser object
+        self.hdlc                    = openhdlc.OpenHdlc()
+        #frame parsing variables
+        self.rxBuffer                = []
+        self.hdlc_flag               = False
+        self.receiving               = False
+        self.xonxoff_escaping        = False
+
+        self.HEADER_LENGTH           = 2
+
+        # to be assigned, callback
+        self.send_to_parser          = None
+
+        # initialize thread
+        super(Serial_RxSnifferThread, self).__init__()
+        self.name                    = 'Serial_RxSnifferThread@{0}'.format(self.serialport)
+        self.start()
+
+    def run(self):
+
+        time.sleep(1)  # let the banners print
+        while self.goOn:
+            try:
+                with self.dataLock:
+                    pleaseConnect = self.pleaseConnect
+
+                if pleaseConnect:
+                    # open serial port
+                    self.serialHandler = serial.Serial(self.serialport, baudrate=self.baudrate)
+
+                    # read byte
+                    while True:
+                        waitingbytes   = self.serialHandler.inWaiting()
+                        if waitingbytes != 0:
+                            c= self.serialHandler.read(waitingbytes)
+                            for byte in c:
+                               self._newByte(byte)
+                            time.sleep(2)
+
+            except serial.SerialException:
+                # mote disconnected, or pyserialHandler closed
+                # destroy pyserial instance
+                print "WARNING: Could not read from serial at \"{0}\".".format(
+                       self.serialport)
+                print "Is device connected?"
+                self.goOn            = False
+                self.serialHandler   = None
+                time.sleep(1)
+
+            except Exception as err:
+                logCrash(self.name, err)
```

Comment:
Should you not destroy the thread in this case as well? i.e. call `self.close()`.